tests/test_result.py

```diff
@@ -375,3 +375,55 @@ async def to_err_async(i: int) -> Result[int, int]:
 # Lambda versions of the same functions, just for test/type coverage
 sq_lambda: Callable[[int], Result[int, int]] = lambda i: Ok(i * i)
 to_err_lambda: Callable[[int], Result[int, int]] = lambda i: Err(i)
+
+
+# Tests for do notation
+
+
+def test_result_do_general() -> None:
+    def resx(is_suc: bool) -> Result[str, int]:
+        return Ok("hello") if is_suc else Err(1)
+
+    def resy(is_suc: bool) -> Result[bool, int]:
+        return Ok(True) if is_suc else Err(2)
+
+    def _get_output(is_suc1: bool, is_suc2: bool) -> Result[float, int]:
+        res_x = resx(is_suc1)
+        res_y = resy(is_suc2)
+
+        out: Result[float, int] = do(
+            Ok(len(x) + int(y) + 0.5)
+            for x in res_x
+            for y in res_y
+        )
```

Comment:
Sorry, this is a bit nitpicky, but could you move the function calls inline here, I think that better reflects usage -- most people wouldn't predefine the values, rather just call them inline in the do expression. Similarly for the async test below.